pkgs/development/compilers/rust/make-rust-platform.nix

```diff
@@ -1,39 +1,46 @@
 {
   lib,
-  buildPackages,
-  callPackages,
   cargo-auditable,
   config,
-  stdenv,
-  runCommand,
-  generateSplicesForMkScope,
   makeScopeWithSplicing',
+  pkgsBuildBuild,
+  pkgsBuildHost,
+  pkgsBuildTarget,
+  pkgsHostHost,
+  pkgsHostTarget,
+  pkgsTargetTarget,
 }@prev:
 
 {
   rustc,
   cargo,
   cargo-auditable ? prev.cargo-auditable,
-  stdenv ? prev.stdenv,
   ...
 }:
 
-(makeScopeWithSplicing' {
-  otherSplices = generateSplicesForMkScope "rustPlatform";
+let
   f =
     self:
     let
       inherit (self) callPackage;
+      callPackages = callPackage ({ callPackages }: callPackages) { };
+      stdenv = callPackage ({ stdenv }: stdenv) { };
+      targetStdenv = callPackage ({ pkgsTargetTarget }: pkgsTargetTarget.stdenv) { };
```

Comment:
If there's a better way of fetching packages from the parent scope, I would love to know.